core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -792,13 +792,32 @@ class PartKeyLuceneIndex(ref: DatasetRef,
     logger.info(s"Refreshed index searchers to make reads consistent for dataset=$ref shard=$shardNum")
   }
 
+
+  private def removeRegexTailDollarSign(regex : String) : String = {
+    // avoid unnecessary calculation when $ is not present at the end of the regex.
+    if (regex.nonEmpty && regex.last == '$' && !regex.dropRight(1).foldLeft(false)((v, c) => !v && c == '\\')) {
```

Comment:
This takes a second to understand-- if you could match with a regex, it would be more straightforward, I think.

Reply:
updated the code with a regex version. Hopefully it is easier to understand.